crates/rolldown_binding/src/types/binding_output_asset.rs

```diff
@@ -1,48 +1,53 @@
-use std::sync::Arc;
+use std::sync::{Arc, Weak};
 
 use napi_derive::napi;
+use rolldown_common::OutputAsset;
 
 use crate::options::plugin::types::binding_asset_source::BindingAssetSource;
 
 #[napi]
 pub struct BindingOutputAsset {
-  inner: Arc<rolldown_common::OutputAsset>,
+  inner: Weak<OutputAsset>,
 }
 
 #[napi]
 impl BindingOutputAsset {
-  pub fn new(inner: Arc<rolldown_common::OutputAsset>) -> Self {
+  pub fn new(inner: Weak<OutputAsset>) -> Self {
     Self { inner }
   }
 
+  fn inner(&self) -> Arc<OutputAsset> {
+    self.inner.upgrade().unwrap()
```

Comment:
Instead of crashing, the js side can set the value to `undefined`, or we throw a panic message "Rolldown has been shutdown, access to this data is no longer possible".